module/VuFind/src/VuFind/RecordDataFormatter/Specs/AbstractBase.php

```diff
@@ -163,6 +164,70 @@ protected function addOptions(string $context, string $field, array $options): ?
             $options = array_merge($options, $contextOptions);
         }
 
+        $includingFilter = array_merge_recursive(
+            $options['filter']['include'] ?? [],
+            $this->getFilterFromConfig($options['filterInclude'] ?? []),
+        );
+        if (!empty($includingFilter)) {
+            $options['multiEnabled'] = false;
+        }
+        foreach ($includingFilter as $lineIdentifierKey => $lineIdentifierValues) {
+            foreach ($lineIdentifierValues as $lineIdentifierValue) {
+                $options['lineOptions'][$lineIdentifierKey][$lineIdentifierValue]['enabled'] = true;
+            }
+        }
+
+        $excludingFilter = array_merge_recursive(
+            $options['filter']['exclude'] ?? [],
+            $this->getFilterFromConfig($options['filterExclude'] ?? []),
+        );
+        foreach ($excludingFilter as $lineIdentifierKey => $lineIdentifierValues) {
+            foreach ($lineIdentifierValues as $lineIdentifierValue) {
+                $options['lineOptions'][$lineIdentifierKey][$lineIdentifierValue]['enabled'] = false;
+            }
+        }
+        unset($options['filterExclude']);
+        unset($options['filterInclude']);
+        unset($options['filter']);
+
+        foreach ($options['extraLineOptions'] ?? [] as $lineOptionSection) {
+            $extraLineOption = $this->config[$lineOptionSection] ?? [];
+            $lineIdentifierKey = $extraLineOption['lineIdentifierKey'] ?? 'label';
+            $lineIdentifierValue = $extraLineOption['lineIdentifierValue'] ?? null;
+            unset($extraLineOption['lineIdentifierKey']);
+            unset($extraLineOption['lineIdentifierValue']);
+            if ($lineIdentifierValue === null) {
+                continue;
+            }
+            $options['lineOptions'][$lineIdentifierKey][$lineIdentifierValue] = array_merge(
+                $options['lineOptions'][$lineIdentifierKey][$lineIdentifierValue] ?? [],
+                $extraLineOption,
+            );
+        }
+        unset($options['extraLineOptions']);
+
         return $options;
     }
+
+    /**
+     * Get filter from config.
+     *
+     * @param array $config Config
+     *
+     * @return array
+     */
+    protected function getFilterFromConfig(array $config): array
```

Comment:
Is this filter logic documented and/or used anywhere?

Reply:
I'm not sure how often those settings have been used in practice, though it may be safe to assume "not widely/extensively." I think it might be better, as you say, to just use existing options rather than adding new ones, even if that might increase configuration verbosity. On the other hand, if you do prefer to do things this way, I think we should add more comments explaining the context/purpose of this method, and perhaps more clearly documenting the options in the .ini file.

I guess another way to think about it: if this is only here to support legacy settings efficiently, I don't think it's worth keeping. If this is generally useful as a convenience to make .ini files more readable, then it probably IS worth keeping, but we should document it better.